### ISSUE TITLE
War retirejs

### ISSUE THREAD

**Issue by tobinbrooke** [CycloneDX/cyclonedx-maven-plugin#219]
We have a java spring maven project that generates a war file. The sbom generated does not include the js libraries. This causes a discrepancy when importing into dependency track Vs dependency check

**Comment by stevespringett** [comment]
Are the js libraries unmanaged dependencies, like in a Maven resources directory?

**Comment by tobinbrooke** [comment]
Correct. Retirejs is able to generate a cyclonedx sbom I am currently looking for a solution which would first generate sbom from retirejs the combine both

**Comment by tobinbrooke** [comment]
The end goal is to use dependency track for both the maven dependencies and js libs

**Comment by stevespringett** [comment]
The Maven plugin will only pick up managed Maven dependencies. If your JavaScript is managed by npm, we have a native npm module that works great. Regardless of how you generate the bom for JavaScript, use the CycloneDX CLI to merge them.

**Comment by tobinbrooke** [comment]
I may have found a solution which involved using another component to generate the js sbom then a command line tool to aggregate the two boms will update with details once verified

**Comment by ashutoshvimal** [comment]
> I may have found a solution which involved using another component to generate the js sbom then a command line tool to aggregate the two boms will update with details once verified

Which another component are you using?

**Comment by ashutoshvimal** [comment]
> I may have found a solution which involved using another component to generate the js sbom then a command line tool to aggregate the two boms will update with details once verified

any progress?
